### ISSUE TITLE
Scrolllock shifts layout.

### ISSUE THREAD

**Issue by sblad** [JedWatson/react-select#5342]
When i set my vertical scrollbars always on with `menuShouldBlockScroll`, there is a padding-right added to the body which shifts entire layout and i get white stripe on the right end of the page. 
Not sure how it should act, but if I set my OS settings to auto hide scrollbars it works okay.
Maybe there should be a prop to opt out from this padding adjustment?

**Comment by Rall3n** [comment]
@sblad Please provide a minimal reproducible example using [CodeSandbox](https://codesandbox.io/s/react-select-v5-sandbox-y5jtm) as requested in the issue template.

**Comment by sblad** [comment]
Sorry. Here is the reproduction: https://codesandbox.io/s/react-select-v5-sandbox-forked-dfof09. As you can see there is no scrollbar on the body, but padding is added anyway shifting entire layout

**Comment by lukasz-bendykowski-fluke** [comment]
Yes, I can confirm that this happens in our product as well. `padding-right` is added to the `body` element when opening a selector. 

![image](https://user-images.githubusercontent.com/40466851/195675784-db6634ec-cacc-422a-9b7e-823faa4ec596.png)

Forcing `padding-right` to `0` in styles dirty-fixed the issue here, but I would prefer a cleaner solution.
